Clock/PyQtPiClock.py

```
@@ -1536,6 +1681,14 @@ def qtstart():
     temptimer.timeout.connect(gettemp)
     temptimer.start(int(1000 * 10 * 60 + random.uniform(1000, 10000)))
 
+    # Fetch RainViewer metadata once at regular intervals (every 10 minutes)
+    metadatatimer = QtCore.QTimer()
+    metadatatimer.timeout.connect(get_rainviewer_metadata)
+    metadatatimer.start(int(1000 * 600 + random.uniform(1000, 5000)))  # 10 minutes
```

Ruff (0.15.20)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)